lib/ansible/modules/windows/win_domain.ps1

```diff
@@ -17,7 +17,7 @@ Function Ensure-Prereqs {
         # NOTE: AD-Domain-Services includes: RSAT-AD-AdminCenter, RSAT-AD-Powershell and RSAT-ADDS-Tools
         $awf = Add-WindowsFeature AD-Domain-Services -WhatIf:$check_mode
         # FUTURE: Check if reboot necessary
-        return $true
+        return (-Not $awf.RestartNeeded)
```

Comment:
So you are returning `$true` when a restart is not needed and then exiting in check mode if the return value here is true. If you are trying to fix the check mode operation to still continue the module check if we've installed the feature and a restart was not required I would have assumed it would only return `$true` if a restart was needed.

Reply:
in the new PR i changed it to $result.reboot_required = $awf.RestartNeeded